text/0000-rustflags.md

```diff
@@ -0,0 +1,356 @@
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC aims to improve the experience of enabling Rust compiler flags for specific crates when building Rust projects
+by adding a new option, `--rustflags <RUSTFLAGS>`. This would have the same effect as `cargo rustc -- <RUSTFLAGS>` but would
+also be available for use by other subcommands such as `bench, build, check, doc, fix, install, package, run` and `test`.
+This allows a Rust project to be built and tested for instance without forcing a new compilation and losing the rustflags that
+were set when invoking `cargo rustc`. This option sets `<RUSTFLAGS>` for the current crate only and not all crates in the
+dependency graph.
+
+# Motivation
+[motivation]: #motivation
+
+Today, there currently exists multiple ways of specifying `RUSTFLAGS` to pass to invocations of the Rust compiler.
+All of the existing ways have the limitation of not being able to specify which invocation of rustc the compiler flag
+is set for. When a Rust developer tries to enable a Rust compiler option for the current crate being built, they will also
+have this compiler option set for all dependencies. With the feature proposed by this RFC, a Rust developer can simply use
+the cargo CLI to pass rustflags for the current crate without having to worry about how other crates in dependency graph
+might be affected.
+
+Some of the existing support for rustflags in Cargo are:
+
+1. `CARGO_ENCODED_RUSTFLAGS` environment variable
+2. `RUSTFLAGS` environment variable
+3. `target.*.rustflags` from the config (.cargo/config)
+4. `target.cfg(..).rustflags` from the config (.cargo/config)
+5. `host.*.rustflags` from the config (.cargo/config) if compiling a host artifact or without `--target`
+6. `build.rustflags` from the config (.cargo/config)
+
+These currently all override one another in the order specified, meaning if `CARGO_ENCODED_RUSTFLAGS` is set, none of the other rustflags
+options are even considered. The same is true if `CARGO_ENCODED_RUSTFLAGS` is not set but `RUSTFLAGS` is, and so on down the list. All of
+these options work in a very similar manner that they are applied for not just the current crate but also all dependencies for said crate
+including transitive dependencies. In the case of the `target.*.rustflags` or `host.*.rustflags` these only apply if the target triple
+specified in the manifest file matches the target triple used as the host or target.
+
+Another supported way of setting rustflags in cargo is the `profile.rustflags` manifest key that can be set in a `Cargo.toml`. This
+works in a slightly different manner than the ways mentioned previously in that it is appended to the set of rustflags calculated from
+the environment variables and cargo config settings. It still has the same limitation in that it is applied to all invocations of rustc
```

Comment:
This is not accurate I believe. One can override a profile of a specific crate[^1] nowadays. With the recently stabilized feature **config-cli**[^2] we can do something like

```
cargo run --config 'profile.dev.package.my-crate.rustflags = ["-C", "instrument-coverage"]'
```

[^1]: https://doc.rust-lang.org/cargo/reference/profiles.html#overrides
[^2]: https://doc.rust-lang.org/cargo/reference/config.html#command-line-overrides

Reply:
Ahh I see. Thank you for the correction. I'll update the RFC to specify that this limitation does not apply to the `profile.rustflags` config setting.